job_templates/ci_job.xml.em

```diff
@@ -229,70 +241,88 @@ setlocal enableDelayedExpansion
 rmdir /S /Q ws workspace "work space"
 
 echo "# BEGIN SECTION: Determine arguments"
-set "PATH=%PATH:"=%"
-set "CI_ARGS=--force-ansi-color --workspace-path %WORKSPACE%"
-if "%CI_BRANCH_TO_TEST%" NEQ "" (
-  set "CI_ARGS=%CI_ARGS% --test-branch %CI_BRANCH_TO_TEST%"
+set "PATH=!PATH:"=!"
+set "CI_ARGS=--force-ansi-color --workspace-path !WORKSPACE!"
+if "!CI_BRANCH_TO_TEST!" NEQ "" (
+  set "CI_ARGS=!CI_ARGS! --test-branch !CI_BRANCH_TO_TEST!"
```

Comment:
Ah, it's an interesting question.  Windows batch files are kind of dumb; they expand any %...% variables at the time the statement is read, not at the time it is executed.  Thus, constructs where you append arguments don't generally work like you expect.  I saw this while I was testing on a local Windows instance.  The workaround for this behavior is to specify `setlocal enableDelayedExpansion`, but that doesn't change the behavior of %...% variables; it just introduces the concept of delayed expansion to !...! variables.  Thus, since we are indeed doing appending on some of the variables, I just switched them all to !...!.  One of the main references I used for this is here: https://ss64.com/nt/delayedexpansion.html